File: frontend/src/tools/purchase_records/recycle_bin/schemas.ts
import { z } from "zod"

export const recycleBinRecordSchema = z.object({
  id: z.string(),
  purchase_date: z.string(),
  name: z.string(),
  amount: z.union([z.number(), z.string(), z.null()]),
  deleted_at: z.string(),
})

export const recycleBinListSchema = z.object({
  data: z.object({
    records: z.array(recycleBinRecordSchema),
  }),
})

export const recycleBinDetailSchema = z.object({
  data: recycleBinRecordSchema,
})

export const recycleBinRestoreSchema = z.object({
  message: z.string(),
})

File: backend/app/modules/purchase_records/recycle_bin/schemas.py
import uuid
from datetime import date, datetime

from sqlmodel import SQLModel


class RecycleBinRecord(SQLModel):
    id: uuid.UUID
    purchase_date: date
    name: str
    amount: float | None
    deleted_at: datetime | None


class RecycleBinListData(SQLModel):
    records: list[RecycleBinRecord]


class RecycleBinListResponse(SQLModel):
    data: RecycleBinListData


class RecycleBinDetailResponse(SQLModel):
    data: RecycleBinRecord


class RecycleBinRestoreResponse(SQLModel):
    message: str
